File: src/rngGame/main/LoadingScreen.java
package rngGame.main;

import javafx.animation.FadeTransition;
import javafx.util.Duration;
import rngGame.visual.AnimatedImage;

public class LoadingScreen extends AnimatedImage {

	private static final LoadingScreen INSTANCE = new LoadingScreen();

	private LoadingScreen() {
		init("./res/gui/Loadingscreen.gif");
		setDisable(true);
		setOpacity(0.0);
	}

	public static LoadingScreen getInstance() {
		return INSTANCE;
	}

		public boolean isInLoadingScreen() {
			return getOpacity() > 0.5;
		}

		public void goIntoLoadingScreen() {
			if (!isInLoadingScreen()) {
				setFitWidth(getImage().getWidth() * WindowManager.getInstance().getScalingFactorX());
				setFitHeight(getImage().getHeight() * WindowManager.getInstance().getScalingFactorY());
				final FadeTransition ft = new FadeTransition(Duration.millis(250.0), this);
				ft.setFromValue(0.0);
				ft.setToValue(1.0);
				ft.play();
			}
		}

		public void goOutOfLoadingScreen() {
			if (isInLoadingScreen()) {
				setFitWidth(getImage().getWidth() * WindowManager.getInstance().getScalingFactorX());
				setFitHeight(getImage().getHeight() * WindowManager.getInstance().getScalingFactorY());
				final FadeTransition ft = new FadeTransition(Duration.millis(250.0), this);
				ft.setFromValue(1.0);
				ft.setToValue(0.0);
				ft.play();
			}
		}

}

File: src/rngGame/main/LoadingScreen.kt
@file:JvmName("LoadingScreen")
package rngGame.main

import javafx.animation.FadeTransition
import javafx.util.Duration
import rngGame.visual.AnimatedImage

object LoadingScreen: AnimatedImage() {

	init {
		init("./res/gui/Loadingscreen.gif")
		isDisable = true
		opacity = 0.0
	}

	fun isInLoadingScreen(): Boolean {
		return opacity > .5
	}

	fun goIntoLoadingScreen() {
		if (!isInLoadingScreen()) {
			fitWidth = image.width * WindowManager.getInstance().scalingFactorX
			fitHeight = image.height * WindowManager.getInstance().scalingFactorY
			val ft = FadeTransition(Duration.millis(250.0), this)
			ft.fromValue = 0.0
			ft.toValue = 1.0
			ft.play()
		}
	}

	fun goOutOfLoadingScreen() {
		if (isInLoadingScreen()) {
			fitWidth = image.width * WindowManager.getInstance().scalingFactorX
			fitHeight = image.height * WindowManager.getInstance().scalingFactorY
			val ft = FadeTransition(Duration.millis(250.0), this)
			ft.fromValue = 1.0
			ft.toValue = 0.0
			ft.play()
		}
	}

}